### DISCUSSION TITLE
Does Avalonia currently support displaying YouTube videos and images in base64 format within a window?

### DISCUSSION THREAD

**Discussion by IronEaterBeast** [AvaloniaUI/Avalonia#15786] [Q&A] [answered]
These are two separate questions, with the YouTube video being an embed code that can be pasted into HTML code to open the video on webpage.

The image is a base64 encoded string.

**Comment by maxkatz6** [comment]
1. On base64 content: you need to convert Base64 to the Stream or byte array on your own, it's simple https://learn.microsoft.com/en-us/dotnet/api/system.convert.frombase64string?view=net-8.0. After what you can create Bitmap instance, if that's an image.

2. On youtube videos: all youtube videos are HTML iframe pages. To show them, you have to use a web browser embedded. You can find some options here https://github.com/AvaloniaCommunity/awesome-avalonia?tab=readme-ov-file#web-browsers

**Reply by IronEaterBeast** [reply]
```
byte[] imageBytes = System.Convert.FromBase64String((string)value!);
using (var ms = new MemoryStream(imageBytes))
{
    return new  Bitmap(ms);
}
```
When I use the statement "new Bitmap(ms)", VS reminds me that it only applies to Windows platform, and I'm not sure if Avalonia has replaced it to make it work on all platforms.

**Reply by thevortexcloud** [reply]
Are you using the right bitmap? Avalonia's bitmap type works fine outside Windows. The System.Drawing.Bitmap type however does not.

**Reply by IronEaterBeast** [reply]
Ok, got it now.

**Comment by thevortexcloud** [comment, marked as answer]
Oh also I forgot to mention, you technically don't need a web browser to play youtube videos. All you need is the stream and some sort of media player. There are several tools available that can extract the streams for you (they do often break though due to youtube changing things), but I wont link them here. Alternatively, if you have just a media player you can also probably directly play the video anyway. VLC for example has built in support for youtube videos and there is an official Avalonia VLC integration.

https://code.videolan.org/videolan/LibVLCSharp/-/tree/master/samples/LibVLCSharp.Avalonia.Sample?ref_type=heads

**Reply by IronEaterBeast** [reply]
Ok, I'll try, thank you!
